src/Libraries/DesignScriptBuiltin/RangeHelpers.cs

```diff
@@ -0,0 +1,378 @@
+﻿using System;
+using Autodesk.DesignScript.Runtime;
+using Builtin.Properties;
+using ProtoCore.DSASM;
+
+namespace ProtoCore.DSASM
+{
+    [SupressImportIntoVM]
+    //Type moved from Protocore and forwarded using type forward attribute.
+    /// <summary>
+    /// Type of range to create.
+    /// </summary>
+    public enum RangeStepOperator
+    {
+        /// <summary>
+        /// Creating entries by stepping using the stepsize.
+        /// </summary>
+        StepSize,
+        /// <summary>
+        /// Create a requested number of entries.
+        /// </summary>
+        Number,
+        /// <summary>
+        /// Approximate step size prioritizing reaching end of the range. 
+        /// </summary>
+        ApproximateSize
+    }
+}
+
+
+namespace Builtin
+{
+    //TODO for now suppress import into DS VM.
+    [SupressImportIntoVM]
+    public class RangeHelpers
+    {
+        //TODO op is specified as a long - even though it is an int.
+        //I guess MSIL engine does not support coercion from long to int,
+        //and all DS ints are longs.
+        /// <summary>
+        /// Generate a range.
+        /// </summary>
+        /// <param name="start">start value</param>
+        /// <param name="end">end value</param>
+        /// <param name="step">step value</param>
+        /// <param name="op">step operation enum see <see cref="RangeStepOperator"/></param>
+        /// <param name="hasStep">has step size been specified</param>
+        /// <param name="hasAmountOp">has amount been specified</param>
+        /// <returns></returns>
+        public static long[] GenerateRangeILInt(long start, long end, double step, long op, bool hasStep, bool hasAmountOp)
+        {
+            return RangeExpressionInt(start, end, step, (int)op, hasStep, hasAmountOp);
+        }
+        public static double[] GenerateRangeILDouble(double start, double end, double step, long op, bool hasStep, bool hasAmountOp)
+        {
+            return RangeExpressionDouble(start, end, step, (int)op, hasStep, hasAmountOp);
+        }
+        //TODO handle alphabetic ranges - overloads or dynamically.
+
+
+
+        internal static long[] RangeExpressionInt(long start, long end, double step, int op, bool hasStep, bool hasAmountOp)
+        {
+            if (hasAmountOp && !hasStep)
+            {
+                throw new ArgumentException(Resources.kNoStepSizeInAmountRangeExpression);
+            }
+
+            return GenerateNumericRangeCoreInt(start, end, step, op, hasStep, hasAmountOp);
```

Comment:
Why not move this code up to the public method? That way we can reduce a few methods, in this case, `RangeExpressionInt` and `RangeExpressionDouble`.

Reply:
These methods are also used from the existing DS range methods- as I started to refactor there - maybe I can combine the methods you have suggested and call the public ones instead .. unsure.

Reply:
Ah, ok, then this is fine.